### ISSUE TITLE
Code cleanup

### ISSUE THREAD

**Issue by carlisia** [vmware-tanzu/velero#2397]
Running a tool like `gosec` finds a bunch of unhandled error and other red flags in the code. We should run it and clean up as much as possible on this and the other Velero related repos.

Note: `gosec` is bundled with `golangci-lint` and you can run it as a plugin from there.

----
Related issues:

https://github.com/vmware-tanzu/velero/issues/2841

**Comment by ironknight78** [comment]
Hey @carlisia ! I'm new here and would like to further #investigate this problem!

**Comment by carlisia** [comment]
Awesome! I assigned this to you. Please let me know if I can guide you with anything.

**Comment by ironknight78** [comment]
Thanks a lot!! I'll start working on it :D and go back for any issues or guidance.

**Comment by ironknight78** [comment]
Which specific packages do you think would be a good place to start @carlisia ?

**Comment by RobbieJVMW** [comment]
@carlisia @ironknight78 if you need another set of fingers on the keyboard let me know.

**Comment by ironknight78** [comment]
@RobbieJVMW Of course!!! Let's tackle this one. There are quite a while messages so we can get this together if you want.
Run the tool and let's pick some :)

**Comment by carlisia** [comment]
Hey @ironknight78 since you were the first to volunteer, why don't you make a list of packages you'll address, and run that by @RobbieJVMW to see if he can do the remainder? We can create a second ticket for @RobbieJVMW.

**Comment by RobbieJVMW** [comment]
Sounds good to me, I've got a test rig setup and cloned the repo.  Currently just looking over all the lines that are flagged with deadcode.   I was considering doing a 'removed dead code' push as the first pass against all the packages.   Any thoughts on that approach ?

**Comment by carlisia** [comment]
> I was considering doing a 'removed dead code' push as the first pass against all the packages. Any thoughts on that approach ?

Yes, let's get this rolling. Please open a ticket for this one and I'll assign it to you. Thank you!

**Comment by carlisia** [comment]
Hey @dbgoytia I am terribly sorry for dropping the ball on this. Would you still be willing to tackle a few packages of the code base and use the linter to clean up the warnings?

**Comment by dbgoytia** [comment]
Hey @carlisia of course! I'll fetch the newest version of the repo and clean up the warnings. I was also a little bit out of the loop for a couple of months on this issue so sorry about that!

**Comment by carlisia** [comment]
No worries. @ me here or on slack if you get stuck and thank you.

**Comment by CypherpunkSamurai** [comment]
Hello everyone 👋🏼 nice to meet you all.

I'm new to open-source and was hoping to contribute to this issue.

Hello Mrs. @carlisia the issue seems to be about fixing lint errors, here's what I understand from reading the issue and related issues:
* We need to clean the lint errors that `gosec` points out.
* Refactor the code and clean it up.

I'm a beginner, please do point out any mistakes :)

**Comment by siddarthpai** [comment]
hey @carlisia , can I take this up if there's anything to do ? 
thanks

**Comment by kaovilai** [comment]
They are [no longer](https://github.com/vmware-tanzu/velero/pull/3913) part of the maintainer roster :)

Gosec is already enabled. https://github.com/vmware-tanzu/velero/blob/83a0badf96dc474053bedfcac00cf836f4faa6bc/.golangci.yaml#L321

I think this issue can close as https://github.com/vmware-tanzu/velero/pull/6008 addressed gosec as part of https://github.com/vmware-tanzu/velero/issues/6024

If you want to suggest more linters enable, feel free and PR and we will review.
